File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        deck = {}
        for i in range(len(nums)):
            if nums[i] in deck:
                return deck[nums[i]], i
            else:
                leftover = target - nums[i]
                deck[leftover] = i
        return -1,-1

File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<int> sol;
        for (size_t i = 0; i < nums.size(); i++) {
            for (size_t j = i +1; j < nums.size(); j++) {
                if (nums.at(i) + nums.at(j) == target) {
                    sol.push_back(i); sol.push_back(j);
                }
            }
        }
        return sol;
    }
};